### PR TITLE
[catch2] Support static build

### LINKED ISSUE #19388: How do I use Catch::Catch2WithMain ?
I was checking the catch2's documentation about their cmake integration (https://github.com/catchorg/Catch2/blob/devel/docs/cmake-integration.md#cmake-targets);

They seem to advise people who don't need a custom main() function to use Catch2::Catch2WithMain, so that's what I went with. 
Only, I seem to have troubles with trying to do so when using catch2 through vcpkg,
my configuration is : 
```
# Tests
find_package(Catch2 CONFIG REQUIRED)
add_executable(tests "src/Test.cxx")
target_link_libraries(tests PRIVATE Catch2::Catch2WithMain)
```

and cmake tells me 
```
CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:534 (_add_executable):
  Target "tests" links to target "Catch2::Catch2WithMain" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
```

However, using Catch2::Catch2 works fine.

Is there a way to use Catch2::Catch2WithMain ?

### PR DESCRIPTION
**Pull request by NancyLi1013**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #19388

Support static build, add static library `Catch2WithMain`.

Note: No feature needs to test.

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]

File: versions/c-/catch2.json [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "d16a7828bceae3fae09c98c1e08071b85bb48973",
+      "version-semver": "2.13.6",
+      "port-version": 2
+    },
+    {
+      "git-tree": "301c50c34de26a8137de74613f7665bf0900b64f",
+      "version-semver": "2.13.7",
+      "port-version": 0
+    },
```
Comment (NancyLi1013) [inline versions/c-/catch2.json:12]:
```suggestion
```

**Review by JackBoosY — APPROVED** [review]

**Comment by Neumann-A** [comment]
> Support static build, add static library Catch2WithMain.

this looks like something which needs to be moved into `manual-link`

**Comment by InfRandomness** [comment]
> > Support static build, add static library Catch2WithMain.
> 
> this looks like something which needs to be moved into `manual-link`

Is that something that needs to be changed in this PR ?
I can probably provide that change since I'm waiting on this to be merged to use Catch2WithMain.

**Comment by JackBoosY** [comment]
@InfRandomness We need to move one of the libraries to _lib/manual-link_.

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**